File: 5.implementation/VT/redsun-service/src/main/java/com/redsun/service/service/ProductsService.java
package com.redsun.service.service;

import com.redsun.service.entities.Products;
import com.redsun.service.entities.Result;

/**
 * Products Service interface
 */
public interface ProductsService {
	
    // Insert.
	Result insert(final Products products) ;

    // Update.
	Result update(final Products products) ;

    // Delete.
	Result delete(final Integer id) ;

	// Get By Id.
	Result getById(final Integer id) ;

    // Exists.
	Result exists(final Integer id) ;

    // Count.
	Result count() ;

	// List for page and filter.
	Result listProductsForPageAndFilter(final int itemsPerPage, final int pageNo, final Products products);

}

File: 5.implementation/NT/redsun/src/main/webapp/static/js/services/ProductsService.js


/**
 * Controller for ${entity.name}
 **/


app.factory('productService', function($http, $rootScope) {
	return {
	
		// Insert.
		add: function(product) {
			var urlServer = $rootScope.makePostURL("/product/insert");
			return $http.post(urlServer, product);
		}
		
		// Update.
		, update: function(productId, product) {
			var urlServer = $rootScope.makePostURL("/product/update/" + productId);
			return $http.put(urlServer, product);
		}
		
		// Delete.
		, delete: function(productId) {
			var urlServer = $rootScope.makePostURL("/product/delete/" + productId);
			return $http.delete(urlServer);
		}
		
		// Get by Id.
		, getById: function(productId) {
			var url = $rootScope.makeGetURL("/product/getbyid/" + productId);
			return $http.get(url);
		}
		
		// List for page and filter.
		, getsForPageAndFilter: function(itemsPerPage, pageNo, product) {
			var url = $rootScope.makeGetURL("/product/list/page/filter/" + itemsPerPage + "/" + pageNo);
			return $http.get(url, { params: product });
		}
		
	}// return.
});